### PR TITLE
test: add tests for CLIPMediaSim

### PR DESCRIPTION
**Pull request by camiloaz**
This PR does the following things:

1. Tests now run on a gpu github runner
2. Removed all the make install commands for specific models and the test steps for specific models since this was just not gonna scale and it just makes the tool development prone to errors or forgetting to add something.
3. Now all tests for all tools run without having to manually add them to the CI.
4. Fixed a test for florence_qa that was wrong, and because we were not running it in CI, we did not realize.
5. Added tests for CLIPMediaSim tool
6. Added documentation for CLIPMediaSim

### CODE REVIEW

**Review by CamiloInx — COMMENTED** [review]
@camiloaz please don't forget to add the docs file showing how to run the tool.

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -5,45 +5,21 @@ on:
 
 jobs:
   run-tests:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-gpu-runner
```
Comment (CamiloInx) [inline .github/workflows/unit-tests.yml:8]: @camiloaz good point. The tests should run on gpu

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -5,45 +5,21 @@ on:
...
+        with:
+          poetry-version: "1.7.1"
 
-      - name: controlnet-aux tests
+      - name: tests
         run: |
-          make install-controlnet-aux
-          make test-controlnet-aux
+          make install
+          make test
```
Comment (CamiloInx) [inline .github/workflows/unit-tests.yml:25]: @camiloaz are you sure that we want to run the tests of all the models? There might be some very heavy models that might not fit on the GPU instance.

File: Makefile [hunk 3]
```diff
@@ -5,47 +5,12 @@ all: install test
...
-	$(POETRY) install -E nsfw-classification --no-interaction
-
-install-controlnet-aux:
-	# Install controlnet_aux dependencies only
-	$(POETRY) install -E controlnet-aux --no-interaction
-
-install-florencev2-qa:
-	# Install florencev2-qa dependencies only
-	$(POETRY) install -E florencev2-qa --no-interaction
-	pip install flash-attn
```
Comment (CamiloInx) [inline Makefile:45]: @camiloaz do you think that it makes sense to install all tools at once instead of allowing users to choose what model to install?

File: pyproject.toml [hunk 4]
```diff
@@ -11,7 +11,7 @@ readme = "README.md"
 "documentation" = "https://github.com/landing-ai/vision-agent-tools"
 
 [tool.poetry.dependencies]
-python = ">=3.9,<4.0"
+python = ">=3.10,<4.0"
```
Comment (CamiloInx) [inline pyproject.toml:14]: Nice change.

File: vision_agent_tools/tools/clip_media_sim.py [hunk 5]
```diff
@@ -54,13 +55,25 @@ def __call__(
         frame_embs = []
         for frame in video:
             inputs = self.processor(images=frame, return_tensors="pt")
-            with torch.no_grad(), torch.autocast(self.device):
+            with torch.autocast(self.device):
                 inputs.to(self.device)
                 outputs = self.model.get_image_features(**inputs)
-            frame_embs.append(outputs.detach())
+                print("outputs", outputs.shape)
```
Comment (CamiloInx) [inline vision_agent_tools/tools/clip_media_sim.py:61]: Remove this print line

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (camiloaz) [reply .github/workflows/unit-tests.yml:26]: I think for now the tests are running fine, so we should not be concerned about that. If we get to a point where the GPU does not fit the model, we can skip those specific tests. But this is much easier for everyone because you don't need to add so much boilerplate to add any tool.

File: Makefile (hunk 3 above)
Comment (camiloaz) [reply Makefile:45]: people can still install specific models, just not with `make` commands. doing this just adds too much things to do for every tool that we add and it's not necessary.

File: vision_agent_tools/tools/clip_media_sim.py (hunk 5 above)
Comment (camiloaz) [reply vision_agent_tools/tools/clip_media_sim.py]: Thanks, fixed.

**Review by dillonalaird — APPROVED** [review]
The code part looks good to me! One thing I was thinking though to make the interface a little easier, instead of passing a timestamp per frame just pass FPS. I think that's a lot easier for the user to handle and is a common video-related argument

**Review by CamiloInx — DISMISSED** [review]

File: docs/clip_media_sim.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+image_path = "path/to/your/image.jpg"
+
+# Path to your video
+video_path = "path/to/your/video.mp4"
+
+# Load the image
+target_image = Image.open(image_path)
+
+# Load the video
+vr = VideoReader('va-generation.mp4', ctx=cpu(0))
```
Comment (CamiloInx) [inline docs/clip_media_sim.md:21]: @camiloaz please change this to `video_path`

Comment (camiloaz) [reply docs/clip_media_sim.md]: Thanks! Fixed

**Comment by camiloaz** [comment]
> The code part looks good to me! One thing I was thinking though to make the interface a little easier, instead of passing a timestamp per frame just pass FPS. I think that's a lot easier for the user to handle and is a common video-related argument

We synced offline and decided to remove the timestamps and return indices. I made the changes for that and added an example to get the timestamp in the documentation.

**Review by dillonalaird — APPROVED** [review]
LGTM

File: vision_agent_tools/tools/clip_media_sim.py [hunk 7]
```diff
@@ -54,13 +71,24 @@ def __call__(
...
                 inputs.to(self.device)
                 outputs = self.model.get_image_features(**inputs)
-            frame_embs.append(outputs.detach())
+            frame_embs.append(outputs.squeeze().detach())
         frame_embs = torch.stack(frame_embs)
-        sims = F.cosine_similarity(frame_embs, target, dim=-1).cpu().numpy()
-        times = np.array(timestamps)
-        output = np.concatenate([times[:, None], sims], axis=1)
+
+        sims = (
```
Comment (dillonalaird) [inline vision_agent_tools/tools/clip_media_sim.py:80]: I forgot this in the file I sent, but these would be the logits right? so we need to still apply softmax

**Review by dillonalaird — APPROVED** [review]

**Review by dillonalaird — APPROVED** [review]
LGTM
